### PR TITLE
feat: gRPC Access Log Service (ALS) sink

### PR DESCRIPTION
**Pull request by zirain**
replace https://github.com/envoyproxy/gateway/pull/3062

### CODE REVIEW

File: internal/xds/translator/accesslog.go [hunk 1]
```diff
@@ -170,6 +174,74 @@ func buildXdsAccessLog(al *ir.AccessLog, forListener bool) []*accesslog.AccessLo
...
+		}
+
+		if len(als.Attributes) > 0 {
+			if attr, err := json.Marshal(als.Attributes); err == nil {
+				md = append(md, &cfgcore.HeaderValue{
+					Key:   "x-accesslog-attr",
+					Value: string(attr),
+				})
+			}
+		}
```
Comment (zirain) [inline internal/xds/translator/accesslog.go:208]: TBH, I didn't understand this very clearly. @arkodg can you recall?

File: test/e2e/base/manifests.yaml [hunk 2]
```diff
@@ -863,3 +863,238 @@ metadata:
...
+        prometheus.io/port: "19001"
+      labels:
+        app: envoy-als
+    spec:
+      containers:
+      - name: envoy-als
+        command:
+        - sh
+        - "-c"
+        - "cp -a /app /app-live && cd /app-live && go  run . "
```
Comment (shawnh2) [inline test/e2e/base/manifests.yaml:1070]:
```suggestion
        - "cp -a /app /app-live && cd /app-live && go run . "
```

**Review by shawnh2 — APPROVED** [review]
LGTM! One small nit.

**Review by AliceProxy — APPROVED** [review]
lgtm, thanks

**Pull request merged**